src/ngMock/angular-mocks.js

```diff
@@ -2006,6 +2009,17 @@ function MockXhr() {
   };
 
   this.abort = angular.noop;
+
+  this.$$events = {};
+  this.addEventListener = function(name, listener) {
+    if (angular.isUndefined(this.$$events[name])) this.$$events[name] = [];
+    this.$$events[name].push(listener);
+  };
```

Comment:
I am kind of confused with this snippet:

When we create a `MockXhr`, we set `this.$$events = {}`.
Then inside the mock `$httpBackend`, right after we create `xhr = new MockXhr()`, we overwrite `xhr.$$events = eventHandlers` (where `eventHandlers` is of type `{[key: string]: function}`).
Then, when `addEventListener` is called on a `MockXhr` instance, we check create an array on `this.$$events[name]` and push the listener.

I would expect the tests to fail, but they obviously don't, so what do I miss ?